### PR TITLE
refactor(condo): INFRA-496 fix importid constraint 

### PR DESCRIPTION
**Pull request by abshnko**
**Problem**
sometimes we have to soft delete a bunch of billing receipt for organization and then create new ones.
With the current (context, importId) constraint it is impossible soft delete and then add a new receipt with the same import id. 

**Solution**
include "deletedAt" condition in the constraint index

### CODE REVIEW

**Review by dkoviazin — APPROVED** [review]

**Comment by Alllex202** [comment]
please add migration with run `yarn makemigrations --empty`

**Review by Alllex202 — COMMENTED** [review]
Any changes in the scheme?

**Comment by abshnko** [comment]
> Any changes in the scheme?

should i add it?

File: apps/condo/migrations/20240924133020-0426_auto_20240924_1030.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    await knex.raw(`
+    BEGIN;
+    --
+    -- [CUSTOM] Set Statement Timeout to some large amount - 25 min (25 * 60 => 1500 sec)
+    --
+    SET statement_timeout = '1500s';
+    --
+    -- Delete old (context, importId) constraint
+    --
+    ALTER TABLE "BillingReceipt" DROP CONSTRAINT "BillingReceipt_importId_9da6acbf_uniq";
```
Comment (pahaz) [inline apps/condo/migrations/20240924133020-0426_auto_20240924_1030.js:14]: Why you don’t change the schema constraints?

**Review by ekabardinsky — APPROVED** [review]

File: apps/condo/migrations/20240925112456-0426_billingreceipt_billingreceipt_unique_context_and_importid.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    await knex.raw(`
+    BEGIN;
+    --
+    -- [CUSTOM] Set Statement Timeout to some large amount - 25 min (25 * 60 => 1500 sec)
+    --
+    SET statement_timeout = '1500s';
+    --
+    -- Delete old (context, importId) constraint
+    --
+    ALTER TABLE "BillingReceipt" ADD CONSTRAINT "BillingReceipt_importId_9da6acbf_uniq" UNIQUE ("context", "importId");
```
Comment (Alllex202) [inline apps/condo/migrations/20240925112456-0426_billingreceipt_billingreceipt_unique_context_and_importid.js:38]: IF NOT EXISTS

File: apps/condo/migrations/20240925112456-0426_billingreceipt_billingreceipt_unique_context_and_importid.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    await knex.raw(`
+    BEGIN;
+    --
+    -- [CUSTOM] Set Statement Timeout to some large amount - 25 min (25 * 60 => 1500 sec)
+    --
+    SET statement_timeout = '1500s';
+    --
+    -- Delete old (context, importId) constraint
+    --
+    ALTER TABLE "BillingReceipt" DROP CONSTRAINT "BillingReceipt_importId_9da6acbf_uniq";
```
Comment (Alllex202) [inline apps/condo/migrations/20240925112456-0426_billingreceipt_billingreceipt_unique_context_and_importid.js:14]: IF EXISTS?

Comment (abshnko) [reply apps/condo/migrations/20240925112456-0426_billingreceipt_billingreceipt_unique_context_and_importid.js]: added, thanks

File: apps/condo/migrations/20240925112456-0426_billingreceipt_billingreceipt_unique_context_and_importid.js (hunk 2 above)
Comment (abshnko) [reply apps/condo/migrations/20240925112456-0426_billingreceipt_billingreceipt_unique_context_and_importid.js:38]: there's no such syntax for constraints, only IF EXISTS

**Pull request merged**